gmso/formats/mol2.py

```diff
@@ -0,0 +1,177 @@
+                        line[1], len(topology.sites) + 1
+                    )
+                )
+                element = None
+            try:
+                charge = float(line[8])
+            except IndexError:
+                warnings.warn(
+                    "No charges were detected for site {} with index {}".format(
+                        line[1], line[0]
+                    )
+                )
+                charge = None
+            atom = Atom(
+                name=line[1],
+                position=position.to("nm"),
+                charge=charge,
+                element=element,
+                residue_label=line[7],
+                residue_index=int(line[6]),
+            )
+            topology.add_site(atom)
+        else:
+            break
+    return line
+
+
+def load_top_bonds(f, topology, **kwargs):
+    """Take a mol2 file section with the heading '@<TRIPOS>BOND' and save to the topology.bonds attribute."""
+    while True:
+        line = f.readline()
+        if "@" not in line and not line == "\n" and line:
+            line = line.split()
+            bond = Bond(
+                connection_members=(
+                    topology.sites[int(line[1]) - 1],
+                    topology.sites[int(line[2]) - 1],
+                )
+            )
+            topology.add_connection(bond)
+        else:
+            break
+    return line
+
+
+def load_top_box(f, topology, **kwargs):
+    """Take a mol2 file section with the heading '@<TRIPOS>FF_PBC' or '@<TRIPOS>CRYSIN' and save to topology.box."""
+    if topology.box:
+        warnings.warn(
+            "This mol2 file has two boxes to be read in, only reading in one with dimensions {}".format(
+                topology.box
+            )
+        )
+        line = f.readline()
+        return line
+    while True:
+        line = f.readline()
+        if "@" not in line and not line == "\n":
+            line = line.split()
+            # TODO: write to box information
+            topology.box = Box(
+                lengths=[float(x) for x in line[0:3]] * u.Å,
+                angles=[float(x) for x in line[3:6]] * u.degree,
+            )
+        else:
+            break
+    return line
+
+
+def parse_record_type_indicator(f, line, topology, site_type):
+    """Take a specific record type indicator from a mol2 file format and save to the proper attribute of a gmso topology.
+
+    Supported record type indicators include Atom, Bond, FF_PBC, and CRYSIN.
+    """
+    supported_rti = {
+        "@<TRIPOS>ATOM\n": load_top_sites,
+        "@<TRIPOS>BOND\n": load_top_bonds,
+        "@<TRIPOS>CRYSIN\n": load_top_box,
+        "@<TRIPOS>FF_PBC\n": load_top_box,
+    }
+    # read in to atom attribute
+    try:
+        line = supported_rti[line](f, topology, site_type=site_type)
+    except KeyError:
+        warnings.warn(
+            "The record type indicator {} is not supported".format(line)
```

Comment:
So what does this do when it returns? Does it still process the lines after it has been returned? The behavior should be included in the warning message I think.

Reply:
It just spits out a warning and skips to the next RTI. 

Adding this to the warning: "Skipping current section and moving to the next RTI header."